google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystem.java

```diff
@@ -195,6 +211,7 @@ public static CloudStorageFileSystem forBucket(
     }
     this.provider = provider;
     this.config = config;
+    this.cloudStorageFileSystem = this;
   }
 
```

Comment:
`this.cloudStorageFileSystem = this` does not work as you probably expect. Each time a new `CloudStorageFileSystem` object is created the static field `cloudStorageFileSystem` will be overridden.

Reply:
Done